component/core/src/main/java/org/exoplatform/social/core/upgrade/SpaceNodeIconsPlugin.java

```diff
@@ -0,0 +1,84 @@
+package org.exoplatform.social.core.upgrade;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+import org.exoplatform.commons.api.persistence.ExoTransactional;
+import org.exoplatform.commons.persistence.impl.EntityManagerService;
+import org.exoplatform.commons.upgrade.UpgradeProductPlugin;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+
+public class SpaceNodeIconsPlugin extends UpgradeProductPlugin {
+
+  private static final Log           LOG              = ExoLogger.getExoLogger(SpaceNodeIconsPlugin.class);
+
+  private static final String        SPACE_NODE_NAMES = "space.node.names";
+
+  private static final String        SPACE_NODE_ICONS = "space.node.icons";
+
+  private final EntityManagerService entityManagerService;
+
+  private final Map<String, String>  spaceNodes       = new HashMap<>();
+
+  public SpaceNodeIconsPlugin(EntityManagerService entityManagerService, InitParams initParams) {
+    super(initParams);
+    this.entityManagerService = entityManagerService;
+    if (initParams.containsKey(SPACE_NODE_NAMES) && initParams.containsKey(SPACE_NODE_ICONS)) {
+      String[] spaceNodeNames = initParams.getValueParam(SPACE_NODE_NAMES).getValue().split(";");
+      String[] spaceNodeIcons = initParams.getValueParam(SPACE_NODE_ICONS).getValue().split(";");
+      for (int i = 0; i < spaceNodeNames.length; i++) {
+        this.spaceNodes.put(spaceNodeNames[i], spaceNodeIcons[i]);
+      }
+    }
+  }
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+    if (spaceNodes.isEmpty()) {
+      LOG.error("Couldn't process upgrade, the parameters '{}' and '{}' are mandatory", SPACE_NODE_NAMES, SPACE_NODE_ICONS);
+      return;
+    }
```

Comment:
Since you don't throw an exception to make the upgrade process retry, it's an expected behavior and at most we will log an information. BTW, you can define this condition in the method `shouldProceedToUpgrade(String newVersion, String previousVersion)`.


```suggestion
  public boolean shouldProceedToUpgrade(String newVersion, String previousVersion) {
    return !spaceNodes.isEmpty();
  }
```

Reply:
fixed